src/sql/src/plan/statement.rs

```diff
@@ -2246,6 +2360,15 @@ fn handle_select(
     })
 }
 
+fn handle_generated_select(scx: &StatementContext, query: String) -> Result<Plan, anyhow::Error> {
+    match parse(query)?.into_element() {
+        Statement::Select(SelectStatement { query, as_of: _ }) => {
+            handle_computed_select(scx, *query)
+        }
+        _ => unreachable!(), // Known to be Select statement.
```

Comment:
totally optional nit: `unreachable!("known to be select statement")` is a kind of cool way to avoid the comment; if you're ever wrong about it being unreachable the error message will print out "known to be a select statement" which can bootstrap the debugging process.